### PR TITLE
[py-tx] Refactor TE.py to api.py

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

TE.net is a dinosaur from another era, and it does a lot of interesting things that need various cleanups (class/module state, java-style class wrapper, naming conventions, directly using GET/POSTS in some commands). We're still a couple of rounds away from getting it into a good shape.

Test Plan
---------

I didn't test this super hard, so it's possible it breaks.
```
$ threatexchange fetch --full # sample data
$ threatexchange experimental-fetch --full # sample data
$ threatexchange label false_positive descriptor 3805925856125337 
```

### CODE REVIEW

File: python-threatexchange/threatexchange/descriptor.py [hunk 1]
```diff
@@ -68,8 +68,7 @@ def is_false_positive(self) -> bool:
     @property
     def is_mine(self) -> bool:
         """This Descriptor is my App's Opinion"""
-        # TODO - come up with a way to do this that doesn't use class state
```
Comment (BarrettOlson) [inline python-threatexchange/threatexchange/descriptor.py:71]: comment is still relevant or no?

File: python-threatexchange/threatexchange/descriptor.py [hunk 2]
```diff
@@ -30,6 +28,8 @@ class ThreatDescriptor(t.NamedTuple):
         ...
     }
     """
+    # TODO - do something smarter than this
```
Comment (BarrettOlson) [inline python-threatexchange/threatexchange/descriptor.py:31]: More context might be good for when we come back and fix this... e.g. 'Class attribute set by CLI, currently pulled from app_token in main.py (pending refactor).'

**Review by BarrettOlson — APPROVED** [review]
LGTM. I don't like the mess around `descriptor.ThreatDescriptor.MY_APP_ID` but can't think of a better alternative atm. 
Added a few comment nits. 

Run the formater before merge? Thanks!

File: python-threatexchange/threatexchange/descriptor.py (hunk 1 above)
Comment (Dcallies) [reply python-threatexchange/threatexchange/descriptor.py:71]: Still using class state :/

Comment (Dcallies) [reply python-threatexchange/threatexchange/descriptor.py:71]: Moved the comment, but still functionally similar

**Pull request merged**